// @vant/cli/dist/package-entry.js
import ActionSheet from '/Users/adamliu/vant/src/action-sheet';
import AddressEdit from '/Users/adamliu/vant/src/address-edit';
import AddressList from '/Users/adamliu/vant/src/address-list';
import Area from '/Users/adamliu/vant/src/area';
import Badge from '/Users/adamliu/vant/src/badge';
import Button from '/Users/adamliu/vant/src/button';
import Calendar from '/Users/adamliu/vant/src/calendar';
import Card from '/Users/adamliu/vant/src/card';
import Cascader from '/Users/adamliu/vant/src/cascader';
import Cell from '/Users/adamliu/vant/src/cell';
import CellGroup from '/Users/adamliu/vant/src/cell-group';
import Checkbox from '/Users/adamliu/vant/src/checkbox';
import CheckboxGroup from '/Users/adamliu/vant/src/checkbox-group';
import Circle from '/Users/adamliu/vant/src/circle';
import Col from '/Users/adamliu/vant/src/col';
import Collapse from '/Users/adamliu/vant/src/collapse';
import CollapseItem from '/Users/adamliu/vant/src/collapse-item';
import ContactCard from '/Users/adamliu/vant/src/contact-card';
import ContactEdit from '/Users/adamliu/vant/src/contact-edit';
import ContactList from '/Users/adamliu/vant/src/contact-list';
import CountDown from '/Users/adamliu/vant/src/count-down';
import Coupon from '/Users/adamliu/vant/src/coupon';
import CouponCell from '/Users/adamliu/vant/src/coupon-cell';
import CouponList from '/Users/adamliu/vant/src/coupon-list';
import DatetimePicker from '/Users/adamliu/vant/src/datetime-picker';
import Dialog from '/Users/adamliu/vant/src/dialog';
import Divider from '/Users/adamliu/vant/src/divider';
import DropdownItem from '/Users/adamliu/vant/src/dropdown-item';
import DropdownMenu from '/Users/adamliu/vant/src/dropdown-menu';
import Empty from '/Users/adamliu/vant/src/empty';
import Field from '/Users/adamliu/vant/src/field';
import Form from '/Users/adamliu/vant/src/form';
import GoodsAction from '/Users/adamliu/vant/src/goods-action';
import GoodsActionButton from '/Users/adamliu/vant/src/goods-action-button';
import GoodsActionIcon from '/Users/adamliu/vant/src/goods-action-icon';
import Grid from '/Users/adamliu/vant/src/grid';
import GridItem from '/Users/adamliu/vant/src/grid-item';
import Icon from '/Users/adamliu/vant/src/icon';
import Image from '/Users/adamliu/vant/src/image';
import ImagePreview from '/Users/adamliu/vant/src/image-preview';
import IndexAnchor from '/Users/adamliu/vant/src/index-anchor';
import IndexBar from '/Users/adamliu/vant/src/index-bar';
import Info from '/Users/adamliu/vant/src/info';
import Lazyload from '/Users/adamliu/vant/src/lazyload';
import List from '/Users/adamliu/vant/src/list';
import Loading from '/Users/adamliu/vant/src/loading';
import Locale from '/Users/adamliu/vant/src/locale';
import NavBar from '/Users/adamliu/vant/src/nav-bar';
import NoticeBar from '/Users/adamliu/vant/src/notice-bar';
import Notify from '/Users/adamliu/vant/src/notify';
import NumberKeyboard from '/Users/adamliu/vant/src/number-keyboard';
import Overlay from '/Users/adamliu/vant/src/overlay';
import Pagination from '/Users/adamliu/vant/src/pagination';
import Panel from '/Users/adamliu/vant/src/panel';
import PasswordInput from '/Users/adamliu/vant/src/password-input';
import Picker from '/Users/adamliu/vant/src/picker';
import Popover from '/Users/adamliu/vant/src/popover';
import Popup from '/Users/adamliu/vant/src/popup';
import Progress from '/Users/adamliu/vant/src/progress';
import PullRefresh from '/Users/adamliu/vant/src/pull-refresh';
import Radio from '/Users/adamliu/vant/src/radio';
import RadioGroup from '/Users/adamliu/vant/src/radio-group';
import Rate from '/Users/adamliu/vant/src/rate';
import Row from '/Users/adamliu/vant/src/row';
import Search from '/Users/adamliu/vant/src/search';
import ShareSheet from '/Users/adamliu/vant/src/share-sheet';
import Sidebar from '/Users/adamliu/vant/src/sidebar';
import SidebarItem from '/Users/adamliu/vant/src/sidebar-item';
import Skeleton from '/Users/adamliu/vant/src/skeleton';
import Sku from '/Users/adamliu/vant/src/sku';
import Slider from '/Users/adamliu/vant/src/slider';
import Step from '/Users/adamliu/vant/src/step';
import Stepper from '/Users/adamliu/vant/src/stepper';
import Steps from '/Users/adamliu/vant/src/steps';
import Sticky from '/Users/adamliu/vant/src/sticky';
import SubmitBar from '/Users/adamliu/vant/src/submit-bar';
import Swipe from '/Users/adamliu/vant/src/swipe';
import SwipeCell from '/Users/adamliu/vant/src/swipe-cell';
import SwipeItem from '/Users/adamliu/vant/src/swipe-item';
import Switch from '/Users/adamliu/vant/src/switch';
import SwitchCell from '/Users/adamliu/vant/src/switch-cell';
import Tab from '/Users/adamliu/vant/src/tab';
import Tabbar from '/Users/adamliu/vant/src/tabbar';
import TabbarItem from '/Users/adamliu/vant/src/tabbar-item';
import Tabs from '/Users/adamliu/vant/src/tabs';
import Tag from '/Users/adamliu/vant/src/tag';
import Toast from '/Users/adamliu/vant/src/toast';
import TreeSelect from '/Users/adamliu/vant/src/tree-select';
import Uploader from '/Users/adamliu/vant/src/uploader';

const version = '2.12.54';

function install(Vue) {
  const components = [
    ActionSheet,
    AddressEdit,
    AddressList,
    Area,
    Badge,
    Button,
    Calendar,
    Card,
    Cascader,
    Cell,
    CellGroup,
    Checkbox,
    CheckboxGroup,
    Circle,
    Col,
    Collapse,
    CollapseItem,
    ContactCard,
    ContactEdit,
    ContactList,
    CountDown,
    Coupon,
    CouponCell,
    CouponList,
    DatetimePicker,
    Dialog,
    Divider,
    DropdownItem,
    DropdownMenu,
    Empty,
    Field,
    Form,
    GoodsAction,
    GoodsActionButton,
    GoodsActionIcon,
    Grid,
    GridItem,
    Icon,
    Image,
    ImagePreview,
    IndexAnchor,
    IndexBar,
    Info,
    List,
    Loading,
    Locale,
    NavBar,
    NoticeBar,
    Notify,
    NumberKeyboard,
    Overlay,
    Pagination,
    Panel,
    PasswordInput,
    Picker,
    Popover,
    Popup,
    Progress,
    PullRefresh,
    Radio,
    RadioGroup,
    Rate,
    Row,
    Search,
    ShareSheet,
    Sidebar,
    SidebarItem,
    Skeleton,
    Sku,
    Slider,
    Step,
    Stepper,
    Steps,
    Sticky,
    SubmitBar,
    Swipe,
    SwipeCell,
    SwipeItem,
    Switch,
    SwitchCell,
    Tab,
    Tabbar,
    TabbarItem,
    Tabs,
    Tag,
    Toast,
    TreeSelect,
    Uploader
  ];

  components.forEach(item => {
    if (item.install) {
      Vue.use(item);
    } else if (item.name) {
      Vue.component(item.name, item);
    }
  });
}

if (typeof window !== 'undefined' && window.Vue) {
  install(window.Vue);
}

export {
  install,
  version,
  ActionSheet,
  AddressEdit,
  AddressList,
  Area,
  Badge,
  Button,
  Calendar,
  Card,
  Cascader,
  Cell,
  CellGroup,
  Checkbox,
  CheckboxGroup,
  Circle,
  Col,
  Collapse,
  CollapseItem,
  ContactCard,
  ContactEdit,
  ContactList,
  CountDown,
  Coupon,
  CouponCell,
  CouponList,
  DatetimePicker,
  Dialog,
  Divider,
  DropdownItem,
  DropdownMenu,
  Empty,
  Field,
  Form,
  GoodsAction,
  GoodsActionButton,
  GoodsActionIcon,
  Grid,
  GridItem,
  Icon,
  Image,
  ImagePreview,
  IndexAnchor,
  IndexBar,
  Info,
  Lazyload,
  List,
  Loading,
  Locale,
  NavBar,
  NoticeBar,
  Notify,
  NumberKeyboard,
  Overlay,
  Pagination,
  Panel,
  PasswordInput,
  Picker,
  Popover,
  Popup,
  Progress,
  PullRefresh,
  Radio,
  RadioGroup,
  Rate,
  Row,
  Search,
  ShareSheet,
  Sidebar,
  SidebarItem,
  Skeleton,
  Sku,
  Slider,
  Step,
  Stepper,
  Steps,
  Sticky,
  SubmitBar,
  Swipe,
  SwipeCell,
  SwipeItem,
  Switch,
  SwitchCell,
  Tab,
  Tabbar,
  TabbarItem,
  Tabs,
  Tag,
  Toast,
  TreeSelect,
  Uploader
};

export default {
  install,
  version
};
